resources/regexes.ts

```diff
@@ -695,7 +724,7 @@ export default class Regexes {
     return anyOfArray(array);
   }
 
-  static parse(regexpString: RegExp | string): RegExp {
+  static parse<T extends ParamsAllTypes>(regexpString: RegExp | string): Regex<ParamsAll[T]> {
```

Comment:
There's the same issue here as with the previous patch.  There are parts of code like [this line](https://github.com/quisquous/cactbot/blob/main/ui/oopsyraidsy/oopsyraidsy.js#L1217) that are calling `Regexes.parse` with a `NetRegexes`-built regex.

The compiler would have yelled if anything in TypeScript was doing this (and so this is more just an issue for the future and a conceptual issue right now).  We can defer changing something in this PR (because there's enough going on), but this feels like something that could get cleaned up afterwards (either by having two separate parse functions, or changing callers appropriately, or both, or something else).